[jonasongg] iP

jonasongg opened the pull request:
### Duke
> “Your mind is for having ideas, not holding them.” – [David Allen](https://dansilvestre.com/productivity-quotes)

Duke, a CLI program, frees your mind of having to remember things you need to do! It's:

- text-based
- extremely easy to learn
- _SUPER FAST_ to use

All you need to do is:

1. Download it [here](https://github.com/jonasongg/ip/releases/tag/v0.1)
2. Open a command window in the folder containing the JAR executable
3. Run the command `java -jar Duke.jar`

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java

public static void main(String[] args) {
    new Duke(Paths.get("data", "duke.txt").toAbsolutePath().toString()).run();
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jonasongg committed:
change greeting message

jonasongg committed:
add scanner for entering commands

jonasongg committed:
create constant for horizontal line

jonasongg committed:
add ability to store text and list them when requested

jonasongg committed:
add Task class

jonasongg committed:
add ability to mark and unmark tasks as done

jonasongg committed:
change to use switch case instead of if else

jonasongg committed:
refactor outputting message into its own function

jonasongg committed:
add Todo class and change output messages

jonasongg committed:
add Deadline class

jonasongg committed:
add Event class

jonasongg committed:
add automated testing and fix inconsistencies

jonasongg committed:
add basic error support

jonasongg committed:
add delete command

jonasongg committed:
close scanner when bye

jonasongg committed:
implement saving data to file

jonasongg committed:
implement loading data to file

jonasongg committed:
implement datetime parsing and displaying

jonasongg committed:
refactor loading from file

jonasongg committed:
refactor storage into its own class

jonasongg committed:
refactor TaskList into its own class

jonasongg committed:
refactor ui into its own class

jonasongg committed:
add Command classes

jonasongg committed:
add parser and integrate with commands

jonasongg committed:
fix index bug for delete and un/mark

jonasongg committed:
add exceptions

jonasongg committed:
add IllegalArgumentsExceptions for all tasks

jonasongg committed:
add exception for out-of-bounds task index

jonasongg committed:
refactor all files to packages

jonasongg committed:
Merge branch 'add-gradle-support'

jonasongg committed:
change build gradle package name

jonasongg committed:
improve parser

jonasongg committed:
add test for command and parser

jonasongg committed:
fix parser bugs and add more parser tests

jonasongg committed:
Add JavaDoc to all files

jonasongg committed:
Tweak code to follow coding standard

jonasongg committed:
Implement find command

jonasongg committed:
Merge branch-A-JavaDoc into master

jonasongg committed:
Merge branch-A-CodingStandards into master

jonasongg committed:
Add CheckStyle and fix style issues

wj331 commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,46 @@
+ * Represents a command to find tasks in the task list based on a keyword.
+ */
+public class FindCommand extends Command {
+    private final String keyword;
```
Should this really be final? Not sure if it is a constant for every FindCommand

wj331 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,59 @@
+ * Represents an event to be used by the Duke chat-bot.
+ */
+public class Event extends Task {
+    protected LocalDateTime from;
```
Maybe consider using a more descriptive name for field? instead of from maybe consider startTime or fromTime

wj331 reviewed (commented):
LGTM, just some nits on naming and final

jonasongg committed:
Add JavaFX UI boilerplate

jonasongg committed:
Refactor: remove UI class, commands return String

jonasongg committed:
Fix exiting Duke

applepiofmyeye commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,40 @@
+                storage.saveData(tasks);
+                System.exit(0);
+            }
+            return command.execute(tasks, storage);
```
Nice abstraction!

applepiofmyeye commented on src/main/java/duke/TaskList.java:
Nice abstraction of the TaskList!

applepiofmyeye reviewed (approved):
Looks good! Very readable and well designed

jonasongg committed:
Add assertions to Parser

There are a few places in Parser where after splitting a string, the length of the resultant array is checked (and it should be 2).

Adding assertions in these places can help make explicit the length of the resultant arrays, and document these assumptions clearly.

Let's continue to use assertions to document such assumptions in the future for ease of maintainability.

jonasongg committed:
Make Event and Deadline fields more descriptive

jonasongg committed:
Reduce nesting in a few files

This commit attempts to improve code quality by making happy path in a few places more prominent by separating it from guard clauses. This also reduces the nesting in these places.

Let's continue to work on the code quality of this project.

jonasongg committed:
Merge pull request #1 from jonasongg/branch-A-Assertions

Add assertions to Parser

jonasongg committed:
Merge branch 'master' into branch-A-CodeQuality

jonasongg committed:
Merge pull request #2 from jonasongg/branch-A-CodeQuality

Improve code quality

jonasongg committed:
Add Timed task class & support for it in Parser

jonasongg committed:
Add support for Timed in Storage

jonasongg committed:
Add units to Timed's string representation

jonasongg committed:
Add Ui screenshot and add title to window

jonasongg committed:
Edit docs README.md

jonasongg committed:
Refactor Parser methods to be shorter

jonasongg committed:
Rename JAR file
